src/LevelParser.cpp

```diff
@@ -0,0 +1,186 @@
+	root.Attribute("height", &height);
+	root.Attribute("tilewidth", &tileSize);
+
+	return LevelDetails(tileSize, sf::Vector2i(width, height));
+}
+
+const TiXmlElement * findNode(const TiXmlElement& root, const std::string & name)
+{
+	for (const TiXmlElement* e = root.FirstChildElement(); e != nullptr; e = e->NextSiblingElement())
+	{
+		if (e->Value() == name)
+		{
+			return e;
+		}
+	}
+	return nullptr;
+}
+
+const TiXmlElement *findNode(const TiXmlElement & root, const std::string & value, const std::string & name)
+{
+	for (const TiXmlElement* e = root.FirstChildElement(); e != nullptr; e = e->NextSiblingElement())
+	{
+		if (e->Value() == value && e->Attribute("name") == name)
+		{
+			return e;
+		}
+	}
+
+	return nullptr;
+}
+
+std::vector<std::vector<int>> decodeTileLayer(const TiXmlElement & tileLayerElement, const LevelDetails & levelDetails)
+{
+	std::vector<std::vector<int>> tileData;
+
+	std::string decodedIDs; //Base64 decoded information
+	const TiXmlElement* dataNode = nullptr; //Store our node once we find it
+	for (const TiXmlElement* e = tileLayerElement.FirstChildElement(); e != nullptr; e = e->NextSiblingElement())
+	{
+		if (e->Value() == std::string("data"))
+		{
+			dataNode = e;
+		}
+	}
+
+	Base64 base64;
+	//Get the text from within the node and use base64 to decode it
+	for (const TiXmlNode* e = dataNode->FirstChild(); e != nullptr; e = e->NextSibling())
+	{
+		const TiXmlText* text = e->ToText();
+		std::string t = text->Value();
+		decodedIDs = base64.base64_decode(t);
+	}
+
+	std::vector<int> layerRow(levelDetails.m_mapSize.x);
+	for (int i = 0; i < levelDetails.m_mapSize.y; ++i)
+	{
+		tileData.push_back(layerRow);
+	}
+
+	for (int rows = 0; rows < levelDetails.m_mapSize.y; ++rows)
+	{
+		for (int cols = 0; cols < levelDetails.m_mapSize.x; ++cols)
+		{
+			tileData[rows][cols] = *((int*)decodedIDs.data() + rows * levelDetails.m_mapSize.x + cols) - 1;
+
+		}
+	}
+
+	return std::move(tileData);
+}
+
+
+int getNumberOfTileLayers(const TiXmlElement & root)
+{
+	int tileLayerCount = 0;
+	for (const TiXmlElement* e = root.FirstChildElement(); e != nullptr; e = e->NextSiblingElement())
+	{
+		if (e->Value() == std::string("layer"))
+		{
+			++tileLayerCount;
+		}
+	}
+	return tileLayerCount;
+}
+
+int getNumberOfTileSets(const TiXmlElement & root)
+{
+	int tileSetCount = 0;
+	const TiXmlElement* tileSetNode = findNode(root, "tileset");
+	assert(tileSetNode);
+	for (const TiXmlElement* e = tileSetNode; e != nullptr; e = e->NextSiblingElement())
+	{
+		if (e->Value() == std::string("tileset") && e->Attribute("name") != std::string("Collidable"))
+		{
+			++tileSetCount;
+		}
+	}
+	return tileSetCount;
+}
+
+void parseTileMap(WorldMap& worldMap, const TiXmlElement & root, const LevelDetails & levelDetails)
+{
+	const int tileLayerCount = getNumberOfTileLayers(root);
+	const TiXmlElement* tileLayerNode = findNode(root, "layer"); //Get the first layer node
+	assert(tileLayerNode);
+	for (int i = 1; i <= tileLayerCount; ++i)
```

Comment:
no reason to compute tileLayerCount up front.  Instead of looping from 1 to tileLayerCount, loop like:

for (const TiXmlElement* e = root.FirstChildElement(); e != nullptr; e = e->NextSiblingElement())
	if (e->Value() == std::string("layer"))